src/software/ai/hl/stp/play/defense/defense_play_fsm.cpp

```diff
@@ -63,52 +63,11 @@ void DefensePlayFSM::defendAgainstThreats(const Update& event)
 
     // Reset tactics if the number of crease defenders or pass defenders
     // we intend to assign has changed
-    setUpCreaseDefenders(static_cast<unsigned int>(crease_defender_assignments.size()));
-    setUpPassDefenders(static_cast<unsigned int>(pass_defender_assignments.size()));
+    DefensePlayBase::setUpCreaseDefenders(static_cast<unsigned int>(crease_defender_assignments.size()));
+    DefensePlayBase::setUpPassDefenders(static_cast<unsigned int>(pass_defender_assignments.size()));
 
-    for (unsigned int i = 0; i < crease_defenders.size(); i++)
-    {
-        auto target = crease_defender_assignments.at(i).target;
-
-        // Determine the number of crease defenders already assigned to the target
-        auto defenders_with_target_count = std::count_if(
-            crease_defender_assignments.begin(), crease_defender_assignments.begin() + i,
-            [&target](const auto& assignment) { return assignment.target == target; });
-
-        // Pick alignment based on how many crease defenders are already assigned to the
-        // target
-        auto alignment = TbotsProto::CreaseDefenderAlignment::CENTRE;
-        if (defenders_with_target_count == 1)
-        {
-            if (event.common.world_ptr->ball().position().y() > 0)
-            {
-                alignment = TbotsProto::CreaseDefenderAlignment::LEFT;
-            }
-            else
-            {
-                alignment = TbotsProto::CreaseDefenderAlignment::RIGHT;
-            }
-        }
-        else if (defenders_with_target_count == 2)
-        {
-            if (event.common.world_ptr->ball().position().y() > 0)
-            {
-                alignment = TbotsProto::CreaseDefenderAlignment::RIGHT;
-            }
-            else
-            {
-                alignment = TbotsProto::CreaseDefenderAlignment::LEFT;
-            }
-        }
-
-        crease_defenders.at(i)->updateControlParams(
-            target, alignment, event.control_params.max_allowed_speed_mode);
-    }
-
-    for (unsigned int i = 0; i < pass_defenders.size(); i++)
-    {
-        pass_defenders.at(i)->updateControlParams(pass_defender_assignments.at(i).target);
-    }
+    DefensePlayBase::setAlignment(event, crease_defender_assignments);
+    DefensePlayBase::updatePassDefenderControlParams(pass_defender_assignments);
```

Comment:
nit: this'll still work even if you don't prefix `DefensePlayBase` since this class inherits from it

Reply:
Yup I know, but I think using the prefix makes it clearer for future maintainers as to how this class relates to its parent class/where it comes from. Thoughts on that?

Reply:
Okay, to me, i think most editors will be able to catch the base class relationship and find the function declaration so I don't strictly think it's necessary. But I think it's definitely more explicit and easier to read. I'm okay if you keep it.